pkg/services/oauthtoken/oauth_token_test.go

```diff
@@ -264,3 +286,202 @@ func (f *FakeAuthInfoStore) UpdateAuthInfo(ctx context.Context, cmd *login.Updat
 func (f *FakeAuthInfoStore) DeleteAuthInfo(ctx context.Context, cmd *login.DeleteAuthInfoCommand) error {
 	return f.ExpectedError
 }
+
+func TestService_TryTokenRefresh(t *testing.T) {
+	type testCase struct {
+		desc      string
+		identity  *authn.Identity
+		oauthInfo *social.OAuthInfo
+		setupEnv  func(cache *localcache.CacheService, authInfoService *authinfotest.FakeService)
+
+		expectHasEntryCalled bool
+		expectedErr          error
+	}
+
+	tests := []testCase{
+		{
+			desc:        "should skip sync when identity is nil",
+			identity:    nil,
+			expectedErr: nil,
+		},
+		{
+			desc:        "should skip sync when identity is not a user",
+			identity:    &authn.Identity{ID: "service-account:1"},
+			expectedErr: nil,
+		},
+		{
+			desc:        "should fail if the user identity cannot be converted to an int",
+			identity:    &authn.Identity{ID: "user:invalidIdentifierFormat"},
+			expectedErr: nil,
+		},
+		{
+			desc:     "should skip if the expiration check has been cached",
+			identity: &authn.Identity{ID: "user:1234"},
+			setupEnv: func(cache *localcache.CacheService, authInfoService *authinfotest.FakeService) {
+				cache.Set("oauth-refresh-token-1234", true, 1*time.Minute)
+			},
+			expectedErr: nil,
+		},
+		{
+			desc:     "should skip if the expiration check has been cached",
+			identity: &authn.Identity{ID: "user:1234"},
+			setupEnv: func(cache *localcache.CacheService, authInfoService *authinfotest.FakeService) {
+				authInfoService.ExpectedError = errors.New("some error")
+			},
+			expectHasEntryCalled: false,
+			expectedErr:          nil,
+		},
+		{
+			desc:     "should not find the user as the user was not logged via OAuth",
+			identity: &authn.Identity{ID: "user:1234"},
+			setupEnv: func(cache *localcache.CacheService, authInfoService *authinfotest.FakeService) {
+				authInfoService.ExpectedError = user.ErrUserNotFound
+			},
+			expectHasEntryCalled: false,
+			expectedErr:          nil,
+		},
+		{
+			desc:     "should refresh the token if the auth module is oauth",
+			identity: &authn.Identity{ID: "user:1234"},
+			setupEnv: func(cache *localcache.CacheService, authInfoService *authinfotest.FakeService) {
+				authInfoService.ExpectedUserAuth = &login.UserAuth{
+					AuthModule: "oauth",
+				}
+			},
+			expectHasEntryCalled: false,
+			expectedErr:          nil,
+		},
+		{
+			desc:     "should skip sync with a valid JWT token and uses an unknown idp",
```

Comment:
The id token in this test  is actually expired I think
```suggestion
			desc:     "should skip sync with an unknown idp",
```